### PR TITLE
[dbg-macro] fix hook KB-H048

### PR DESCRIPTION
**Pull request by mathbunnyru**
Specify library name and version:  **lib/1.0**

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by uilianries — DISMISSED** [review]
LGTM

**Comment by mathbunnyru** [comment]
dbg-macro doesn't work with gcc 4.9, so I have to check the compiler.
This package is header-only, so I can't check the compiler in package itself.
But I can check the compiler in test_package.

I'm not sure if this is a good way to go, and if someone knows a better way, that would be great.

**Comment by mathbunnyru** [comment]
Need help here

File: recipes/dbg-macro/all/test_package/conanfile.py [hunk 1]
```diff
@@ -11,6 +12,19 @@ def build(self):
         cmake.configure()
         cmake.build()
 
+    def configure(self):
```
Comment (uilianries) [inline recipes/dbg-macro/all/test_package/conanfile.py:15]: Now I see! YOu can't restrict anything in test package, you have to move it to the recipe package instead.

**Review by uilianries — CHANGES_REQUESTED** [review]

File: recipes/dbg-macro/all/test_package/conanfile.py (hunk 1 above)
Comment (mathbunnyru) [reply recipes/dbg-macro/all/test_package/conanfile.py:15]: But checking for compiler doesn't work for header-only recipes, does it?

I was getting an error:
```
ERROR: dbg-macro/0.4.0: Error in configure() method, line 24
	if self.settings.compiler == "gcc" and tools.Version(self.settings.compiler.version) < "5":
	ConanException: 'settings.compiler' doesn't exist
'settings' possible configurations are none
```

Comment (uilianries) [reply recipes/dbg-macro/all/test_package/conanfile.py:15]: It works if you add to settings. As it's a header-only, the package ID can be unique, does not matter which settings are declared.

Comment (mathbunnyru) [reply recipes/dbg-macro/all/test_package/conanfile.py]: Thanks.

**Review by uilianries — DISMISSED** [review]
LGTM

File: recipes/dbg-macro/all/conanfile.py [hunk 2]
```diff
@@ -21,6 +22,19 @@ def source(self):
         extracted_dir = self.name + "-" + self.version
         os.rename(extracted_dir, self._source_subfolder)
 
+    def configure(self):
+        minimal_cpp_standard = "11"
+        if self.settings.get_safe("compiler.cppstd"):
+            tools.check_min_cppstd(self, minimal_cpp_standard)
+
+        if tools.Version(self.settings.compiler.version) < "5":
```
Comment (ericLemanissier) [inline recipes/dbg-macro/all/conanfile.py:30]: What happens on apple-clang and msvc here ?

Comment (uilianries) [reply recipes/dbg-macro/all/conanfile.py:30]: True, but apple-clang is hard to use old versions, as OSX imposes new versions and msvc people usually run 2015 or higher. GCC is another history, people still use old versions for embedded stuff for instance. Anyway, I'm not opposed to adding both.

Comment (mathbunnyru) [reply recipes/dbg-macro/all/conanfile.py]: Lost gcc if during my local debug on mac with apple-clang.
Added it back.
Good note, @ericLemanissier !

**Review by uilianries — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Pull request merged**